File: IntegrationTests/Parser/TypeParsing/FragmentUsageTests/DerivedClassWithValidFragmentUsage.cs
using InjectionCop.Fragment;

namespace InjectionCop.IntegrationTests.Parser.TypeParsing.FragmentUsageTests
{
  public class DerivedClassWithValidFragmentUsage : BaseClass
  {
    public void Foo ([Fragment("ValidFragmentType")] int parameter1, string parameter2)
    {
    }
  }
}

File: IntegrationTests.VB/Parser/TypeParsing/FragmentUsageTests/DerivedClassWithValidFragmentUsage.vb
Imports InjectionCop.Fragment
Imports System

Namespace Parser.TypeParsing.FragmentUsageTests
	Public Class DerivedClassWithValidFragmentUsage
		Inherits BaseClass

    Public Overrides Sub Foo(<Fragment("ValidFragmentType")> parameter1 As Integer, parameter2 As String)
    End Sub
	End Class
End Namespace
